tests/testthat/test-cert.R

```diff
@@ -1,149 +1,149 @@
-havingHttpRecorder <- function(expr) {
-  withr::local_options(rsconnect.http = httpTestRecorder)
-  eval(expr)
-}
-
 test_that("certificates can be saved", {
-  havingFakeConfig(
-    havingHttpRecorder({
-      # add a server with a sample certificate
-      addServer(url = "https://localhost:4567/",
-                name = "cert_test_a",
-                certificate = test_path("certs/sample.crt"),
-                quiet = TRUE)
-
-      # read it back
-      info <- serverInfo("cert_test_a")
-
-      # compare with the contents of the cert we read
-      certLines <- paste(readLines(test_path("certs/sample.crt")), collapse = "\n")
-      expect_equal(certLines, info$certificate)
-    })
+  local_temp_config()
+  local_http_recorder()
+
+  # add a server with a sample certificate
+  addServer(
+    url = "https://localhost:4567/",
+    name = "cert_test_a",
+    certificate = test_path("certs/sample.crt"),
+    quiet = TRUE
   )
+
+  # read it back
+  info <- serverInfo("cert_test_a")
+
+  # compare with the contents of the cert we read
+  certLines <- paste(readLines(test_path("certs/sample.crt")), collapse = "\n")
+  expect_equal(certLines, info$certificate)
 })
 
 test_that("certificates can be added", {
-  havingFakeConfig(
-    havingHttpRecorder({
-      # add a server without a certificate
-      addServer(url = "https://localhost:4567/",
-                name = "cert_test_b",
-                quiet = TRUE)
-
-      # add the certificate
-      addServerCertificate(name = "cert_test_b",
-                           certificate = "certs/sample.crt",
-                           quiet = TRUE)
-
-      # read the server info
-      info <- serverInfo("cert_test_b")
-
-      # see if the cert we added is there
-      certLines <- paste(readLines("certs/sample.crt"), collapse = "\n")
-      expect_equal(certLines, info$certificate)
-    })
+  local_temp_config()
+  local_http_recorder()
+
+  # add a server without a certificate
+  addServer(
+    url = "https://localhost:4567/",
+    name = "cert_test_b",
+    quiet = TRUE
+  )
+
+  # add the certificate
+  addServerCertificate(
+    name = "cert_test_b",
+    certificate = "certs/sample.crt",
```

Comment:
Another opportunity for `test_path`? (multiple examples)

Reply:
Good point.